mat64/dense_arithmetic.go

```diff
@@ -124,12 +124,21 @@ func (m *Dense) Norm(ord float64) float64 {
 func (m *Dense) Add(a, b Matrix) {
 	ar, ac := a.Dims()
 	br, bc := b.Dims()
-
 	if ar != br || ac != bc {
 		panic(ErrShape)
 	}
 
+	aMat, _ := untranspose(a)
```

Comment:
I don't know what you mean. I'm pretty sure the untranspose code is correct. The rest of the Add function uses the original matrices a and b, rather than aMat and bMat. So, the actual code path followed is the final one as a and b are neither RawMatrixers or Vectorers. A better implementation of Add would be to use aMat so that we can get the RawMatrixer fast path, and take into account the actual transposes. I didn't do that in order to minimize the surface of this commit.